projects/client/RabbitMQ.Client/src/client/impl/Connection.cs

```diff
@@ -1047,16 +1049,7 @@ public void MaybeStartHeartbeatTimers()
 
         public void StartMainLoop(bool useBackgroundThread)
         {
-            var taskName = "AMQP Connection " + Endpoint;
-
-#if NETFX_CORE
-            Task.Factory.StartNew(this.MainLoop, TaskCreationOptions.LongRunning);
-#else
-            var mainLoopThread = new Thread(MainLoop);
-            mainLoopThread.Name = taskName;
-            mainLoopThread.IsBackground = useBackgroundThread;
-            mainLoopThread.Start();
-#endif
```

Comment:
NETFX_CORE code is redundant as the compilation symbol is never created (haven't removed other instances). Moved to Task.Run instead of new Thread, which uses the ThreadPool and unlock turning the MainLoop into an async Task (would fit well with System.IO.Pipelines which I have already started working on in another branch).